mailpoet/lib/Automation/Engine/Data/StepRunArgs.php

```diff
@@ -150,4 +155,12 @@ public function getFieldValue(string $key) {
     }
     return $value;
   }
+
+  public function getRunNumber(): int {
+    return $this->runNumber;
+  }
+
+  public function isFirstRun(): bool {
+    return $this->runNumber === 1;
+  }
```

Comment:
nitpick: not sure how useful this method is. We only need it in delay action and atm to me this seems more like extracting delay action logic than providing a general useful StepRunArgs method. Not sure though yet and not a big deal as well :thinking:

Reply:
@websupporter My thinking here was that usually actions that need to schedule "progress" will differentiate between "first" and "other" run. For instance, the send email step will schedule sending on the first run, then schedule progress, and on any other run, it will just check if the email was already sent, and if not, it can mark it as failed, for instance (we will also inform the automation immediately after email sending to avoid "polling", but I discussed with JM scheduling a "progress" to a week or a month to catch when no email was sent at all).

Reply:
That makes sense. Thank you for clarifying.